myapp/app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+
+  def show
+    @task = Task.find(params[:id])
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    @task = Task.new(task_params)
+    if @task.save
+      flash[:notice] = 'Task was successfully created.'
+      redirect_to @task
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @task = Task.find(params[:id])
```

Comment:
```ruby
class TasksController < ApplicationController
  before_action :set_task, only: [:show, :edit, :update, :destroy]

  def show
    # @task is set by the before_action :set_task
  end

  def edit
    # @task is set by the before_action :set_task
  end

  def update
    if @task.update(task_params)
    ////
  end

  def destroy
    @task.destroy
     ////
  end

  private

  def set_task
    @task = Task.find(params[:id])
  end
```

Perhaps it is a matter of taste, but I thought it would be a good idea to put them together since there is some duplication of processing!

Reply:
Thank you！ I changed it